userspace/libsinsp/events/sinsp_events.cpp

```diff
@@ -81,27 +81,36 @@ std::unordered_set<std::string> libsinsp::events::event_set_to_names(
 				resolved_generic = true;
 			}
 		} else {
-			events_names_set.insert(scap_get_event_info_table()[ev].name);
+			events_names_set.insert(g_infotables.m_event_info[ev].name);
 		}
 	}
 	return events_names_set;
 }
 
-// todo(jasondellaluce): think about how we can handle well PPME_ASYNCEVENT_E
 libsinsp::events::set<ppm_event_code> libsinsp::events::names_to_event_set(
         const std::unordered_set<std::string>& events) {
 	std::unordered_set<std::string> remaining_events = events;
 	libsinsp::events::set<ppm_event_code> ppm_event_set;
 
 	// Main loop, on events (ie: non generic events)
 	for(int ppm_ev = 2; ppm_ev < PPM_EVENT_MAX; ++ppm_ev) {
-		const char* ppm_ev_name = scap_get_event_info_table()[ppm_ev].name;
+		const char* ppm_ev_name = g_infotables.m_event_info[ppm_ev].name;
 		if(events.find(ppm_ev_name) != events.end()) {
 			ppm_event_set.insert((ppm_event_code)ppm_ev);
 			remaining_events.erase(ppm_ev_name);
 		}
 	}
 
+	// Secondary loop, on specific async event names, as filled by sinsp
```

Comment:
Here is the primary logic of the patch: given that the `g_infotables` now contains entries for each async event name, we can finally match them too!
Of course we always store `PPME_ASYNCEVENT_E` as `ppm_ev` idx.

Reply:
oh nice and I remember how we all worked on this stuff a good while ago ...